src/main/scala/mesosphere/marathon/state/PathId.scala

```diff
@@ -97,10 +118,46 @@ case class PathId(path: Seq[String], absolute: Boolean = true) extends Ordered[P
   override val hashCode: Int = scala.util.hashing.MurmurHash3.productHash(this)
 }
 
+case class AbsolutePathId(path: Seq[String]) extends PathId {
```

Comment:
OMG. This is so much easier. I actually would love the relative path ids be used only to construct absolute paths.